src/components/patterns/algolia-course-list.tsx

```diff
@@ -0,0 +1,127 @@
+"use client";
+
+import algoliasearch from "algoliasearch/lite";
+import {useHits,usePagination, Configure} from "react-instantsearch";
+import {InstantSearchNext} from "react-instantsearch-nextjs";
+import {useEffect, useMemo, useState} from "react";
+import {Hit as HitType} from "instantsearch.js";
+import SummerCourse from "@components/algolia-results/summer-course/summer-course";
+import useFavorites from "@lib/hooks/useFavorites";
+import { useSearchParams } from "next/navigation";
+import { AlgoliaHit } from "@components/algolia-results/default";
+
+type Props = {
+  appId: string
+  searchIndex: string
+  searchApiKey: string
+}
+
+type FavoriteItem = {
+  uuid: string;
+  title: string;
+  path?: string; 
+  units: number;
+}
+
+type FavoritesList = FavoriteItem[];
+
+const AlgoliaCourseList = ({appId, searchIndex, searchApiKey}: Props) => {
+  const searchClient = useMemo(() => algoliasearch(appId, searchApiKey), [appId, searchApiKey])
+  const [urlParams, setUrlParams] = useState<FavoritesList>([]);
+  const { favs, addFav } = useFavorites()
+  const searchParams = useSearchParams();
+
+  useEffect(() => {
+    const favParamData = searchParams.get("fav");
+    if (favParamData) {
+      try {
+        const favData = favParamData.split(",").map(fav => {
+          const [uuid, title, unitsStr] = fav.split(":");
+          const units = Number(unitsStr);
+          return { uuid, title, units };
+        })
+        setUrlParams(favData);
+
+        favData.forEach(({ uuid, title, units }) => {
+          if (!favs.some(fav => fav.uuid === uuid)) {
+            addFav(uuid, title, "", units);
+          }
+        });
+      } catch (error) {
+        throw new Error(`Error parsing favData: ${error}`);
+      }
+    }
+  }, [searchParams, addFav, favs]);
+
+  const itemUuids = urlParams.map(item => item.uuid) || favs.map(item => item.uuid)
```

Comment:
If I choose some items, and go to view my favorites, I don't get the list of only my favorites. I think you want something like this:
`const itemUuids = urlParams.length > 0  ? urlParams.map(item => item.uuid) : favs.map(item => item.uuid)`